streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/KTableKTableJoinNode.java

```diff
@@ -56,9 +64,39 @@
               thisJoinSide,
               otherJoinSide);
 
+        this.keySerde = keySerde;
+        this.valueSerde = valueSerde;
         this.joinThisStoreNames = joinThisStoreNames;
         this.joinOtherStoreNames = joinOtherStoreNames;
-        this.materializedInternal = materializedInternal;
+        this.queryableStoreName = queryableStoreName;
+        this.storeBuilder = storeBuilder;
+    }
+
+    public Serde<K> keySerde() {
+        return keySerde;
+    }
+
+    public Serde<VR> valueSerde() {
+        return valueSerde;
+    }
+
+    public String[] joinThisStoreNames() {
+        return joinThisStoreNames;
+    }
+
+    public String[] joinOtherStoreNames() {
+        return joinOtherStoreNames;
+    }
+
+    public String queryableStoreName() {
+        return queryableStoreName;
```

Comment:
nit: we do not need to save another reference of queryableStoreName at this class, instead we can add a getter for the field with the same name in `KTableKTableJoinMerger` public and just return `joinMergeProcessorParameters.queryableStoreName()` here.